### PR TITLE
[NewExe] cache infer_meta for program_interpreter

### PR DESCRIPTION
**Pull request by zhangbo9674**
### PR types

Others
### PR changes

Others
### Description

为 Program interpreter 添加 infer shape 的缓存机制
Pcard-67164

### CODE REVIEW

**Review by yuanlehome — APPROVED** [review]
LGTM for CI-Inference

**Review by From00 — APPROVED** [review]
LGTM

File: paddle/fluid/framework/operator.cc [hunk 1]
```diff
@@ -607,6 +607,28 @@ RuntimeInferShapeContext::GetPhiDefaultKernelSignature() const {
 
 void RuntimeInferShapeContext::SetSkipLoD(bool skip) { can_skip_lod_ = skip; }
 
+bool RuntimeInferShapeContext::HasRuntimeAttributes() const {
+  bool is_runtime = false;
+  if (phi::DefaultKernelSignatureMap::Instance().Has(op_.Type())) {
+    auto phi_kernels = phi::KernelFactory::Instance().SelectKernelMap(
+        GetPhiDefaultKernelSignature()->name);
```
Comment (zhiqiu) [inline paddle/fluid/framework/operator.cc:614]: Is default siganature enough?

**Review by zhiqiu — APPROVED** [review]
LGTM for const_cast

**Pull request merged**